homeassistant/components/mikrotik/__init__.py

```diff
@@ -1 +1,390 @@
+        except (librouteros.exceptions.TrapError,
+                librouteros.exceptions.MultiTrapError,
+                librouteros.exceptions.ConnectionError):
+            _LOGGER.info("Mikrotik %s: Not a CAPsMAN controller. Trying "
+                         "local wireless interfaces.", (host))
+            capsman_exist = False
+        try:
+            wireless_exist = self._client[host](
+                cmd=MIKROTIK_SERVICES[WIRELESS])
+        except (librouteros.exceptions.TrapError,
+                librouteros.exceptions.MultiTrapError,
+                librouteros.exceptions.ConnectionError):
+            wireless_exist = False
+
+        if not wireless_exist and not capsman_exist \
+                or polling_method == 'ip':
+            _LOGGER.info(
+                "Mikrotik %s: Wireless adapters not found. Try to "
+                "use DHCP lease table as presence tracker source. "
+                "Please decrease lease time as much as possible",
+                host)
+        if polling_method:
+            _LOGGER.info("Mikrotik %s: Manually selected polling method %s",
+                         host, polling_method)
+            device_tracker = polling_method
+        else:
+            if capsman_exist:
+                device_tracker = CAPSMAN
+            elif wireless_exist:
+                device_tracker = WIRELESS
+            else:
+                device_tracker = DHCP
+            _LOGGER.info("Mikrotik %s: Using device_tracker method %s",
+                         host, device_tracker)
+        self._hosts[host][DEVICE_TRACKER] = device_tracker
+
+    def connect_to_device(self, host):
+        """Connect to Mikrotik method."""
+        import librouteros
+        import time
+        tick = 0
+        while self._hosts[host][CONNECTING] and tick < 60:
+            time.sleep(1)
+            tick += 1
+        if self._hosts[host][CONNECTED]:
+            return
+        self._hosts[host][CONNECTING] = True
+        self._hosts[host][CONNECTED] = False
+        _LOGGER.debug("[%s] Connecting to Mikrotik device.", host)
+        config = self._hosts[host]['config']
+        self.config_kwargs(config)
+        try:
+            self._client[host] = librouteros.connect(
+                config[CONF_HOST], config[CONF_USERNAME],
+                config.get(CONF_PASSWORD, ''),
+                **self._hosts[host]['kwargs'])
+        except (librouteros.exceptions.TrapError,
+                librouteros.exceptions.MultiTrapError,
+                librouteros.exceptions.ConnectionError) as api_error:
+            _LOGGER.error(
+                "Mikrotik error for device %s. "
+                "Connection error: %s", host, api_error)
+            self._hosts[host][CONNECTING] = False
+            self._hosts[host][CONNECTED] = False
+            self._client[host] = None
+            return False
+
+        host_name = (self._client[host](
+            cmd=MIKROTIK_SERVICES[IDENTITY]))[0]['name']
```

Comment:
This is hard to read. Maybe cache the client in a local variable first to make it more readable?

Reply:
Okay, making a change to use client and cmd local variables